File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
            vector<int> alp(26, 0);
        for (int i = 0; i < keymap.size(); i++)
        {
            for (int j = 0; j < keymap[i].size(); j++)
            {
                int place = (keymap[i][j] - 0) % 65;
                if (alp[place] == 0)  alp[place] = j+1;
                else alp[place] = min(alp[place], j+1);
            }
        }

                for (int i = 0; i < targets.size(); i++)
        {
            int sum = 0;
            bool chk = false;
            for (int j = 0; j < targets[i].size(); j++)
            {
                if (alp[(targets[i][j] - 0) % 65] == 0)
                {
                    chk = true;
                    break;
                }
                sum += alp[(targets[i][j] - 0) % 65];
            }
            if (chk) answer.push_back(-1);
            else answer.push_back(sum);
        }
    return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cs
using System;
using System.Linq;
public class Solution {
    public int[] solution(string[] keymap, string[] targets) {
        int[] answer = new int[targets.Length];
int index = 0;


foreach(var item in targets)
{
    foreach(var word in item)
    {
        bool chk = true;
        int min = int.MaxValue;
        foreach(var key in keymap)
        {
            if(key.Contains(word))
            {
                min = Math.Min(key.IndexOf(word),min);
            }
        }
        if(min == int.MaxValue)
        {
            answer[index] = -1;
            chk = false;
            break;
        }
        else answer[index] += (min+1);
        if (!chk) break;
    }
    index++;
}
        
        return answer;
    }
}